src/test/basics/RangeSet_test.cpp

```diff
@@ -78,39 +76,70 @@ class RangeSet_test : public beast::unit_test::suite
     }
 
     void
-    testSerialization()
+    testFromString()
     {
+        testcase("fromString");
 
-        auto works = [](RangeSet<std::uint32_t> const & orig)
-        {
-            std::stringstream ss;
-            boost::archive::binary_oarchive oa(ss);
-            oa << orig;
-
-            boost::archive::binary_iarchive ia(ss);
-            RangeSet<std::uint32_t> deser;
-            ia >> deser;
-
-            return orig == deser;
-        };
-
-        RangeSet<std::uint32_t> rs;
-
-        BEAST_EXPECT(works(rs));
-
-        rs.insert(3);
-        BEAST_EXPECT(works(rs));
-
-        rs.insert(range(7u, 10u));
-        BEAST_EXPECT(works(rs));
+        RangeSet<std::uint32_t> set;
 
+        BEAST_EXPECT(!from_string(set, ""));
+        BEAST_EXPECT(boost::icl::length(set) == 0);
+
+        BEAST_EXPECT(!from_string(set, "#"));
+        BEAST_EXPECT(boost::icl::length(set) == 0);
+
+        BEAST_EXPECT(!from_string(set, ","));
+        BEAST_EXPECT(boost::icl::length(set) == 0);
+
+        BEAST_EXPECT(!from_string(set, ",-"));
+        BEAST_EXPECT(boost::icl::length(set) == 0);
+
+        BEAST_EXPECT(!from_string(set, "1,,2"));
+        BEAST_EXPECT(boost::icl::length(set) == 0);
+
+        set.clear();
+        BEAST_EXPECT(from_string(set, "1"));
+        BEAST_EXPECT(boost::icl::length(set) == 1);
+        BEAST_EXPECT(boost::icl::first(set) == 1);
+
+        set.clear();
+        BEAST_EXPECT(from_string(set, "1,1"));
+        BEAST_EXPECT(boost::icl::length(set) == 1);
+        BEAST_EXPECT(boost::icl::first(set) == 1);
+
+        set.clear();
+        BEAST_EXPECT(from_string(set, "1-1"));
+        BEAST_EXPECT(boost::icl::length(set) == 1);
+        BEAST_EXPECT(boost::icl::first(set) == 1);
+
+        set.clear();
+        BEAST_EXPECT(from_string(set, "1,4-6"));
+        BEAST_EXPECT(boost::icl::length(set) == 4);
+        BEAST_EXPECT(boost::icl::first(set) == 1);
+        BEAST_EXPECT(boost::icl::contains(set, 4));
+        BEAST_EXPECT(boost::icl::last(set) == 6);
```

Comment:
Since this is a small set, and to be thorough, I think it would be useful to test that all the expected values are there, and that one or two unexpected values are not.
```
BEAST_EXPECT(!boost::icl::contains(set, 2));
BEAST_EXPECT(boost::icl::contains(set, 5));
````